### PR TITLE
fix: handle subaccount case in gaec

### PR DESCRIPTION
**Pull request by aashishmalik**
## What are the changes introduced in this PR?

handle subaccount is false but loginCustomerId is present in config

## What is the related Linear task?

Resolves INT-XXX

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3876/test-report.html

File: src/v0/destinations/google_adwords_enhanced_conversions/transform.js [hunk 1]
```diff
@@ -84,9 +84,14 @@ const responseBuilder = async (metadata, message, { Config }, payload) => {
     response.headers['login-customer-id'] = filteredLoginCustomerId;
   }
 
-  if (loginCustomerId) {
-    const filteredLoginCustomerId = removeHyphens(loginCustomerId);
-    response.headers['login-customer-id'] = filteredLoginCustomerId;
+  if (
+    (isDefined(Config.subAccount) && Config.subAccount === true) ||
```
Comment (krishna2020) [inline src/v0/destinations/google_adwords_enhanced_conversions/transform.js:88]: your new UI shouldn't depend on old UI field of `subAccount` but should depend on new UI field `configData`

Comment (krishna2020) [reply src/v0/destinations/google_adwords_enhanced_conversions/transform.js:88]: And raise PR to hotfix branch

Comment (aashishmalik) [reply src/v0/destinations/google_adwords_enhanced_conversions/transform.js]: `{
                rudderAccountId: '25u5whFH7gVTnCiAjn4ykoCLGoC',
                listOfConversions: ['Page View', 'Product Added'],
                authStatus: 'active',
                subAccount: false,
                configData: '{"customerId": "1234567890", "loginCustomerId": "12345"}',
                customerId: '1234567890',
                loginCustomerId: '12345',
              },`
              
This is how the config will look for edge case when subAccount is flase but config has field loginCustomerId. ConfigData will have loginCustomerId but need to depend on subAccount if it to be used or not.
 
 for new connections, configData will be  {"customerId": "1234567890", "loginCustomerId": "12345"} and no key `subAccount` in Config so use login customerId

Comment (krishna2020) [reply src/v0/destinations/google_adwords_enhanced_conversions/transform.js]: In that case, if `configDetails` object is not null, you shouldn't execute old logic - right now code is messed up where both old & new flows will be executed.

Comment (aashishmalik) [reply src/v0/destinations/google_adwords_enhanced_conversions/transform.js]: `{
                rudderAccountId: '25u5whFH7gVTnCiAjn4ykoCLGoC',
                listOfConversions: ['Page View', 'Product Added'],
                authStatus: 'active',
                subAccount: false,
                configData: '{"customerId": "1234567890", "loginCustomerId": "12345"}',
                customerId: '1234567890',
                loginCustomerId: '12345',
              },`
              
This is how the config will look for edge case when subAccount is flase but config has field loginCustomerId. ConfigData will have loginCustomerId but need to depend on subAccount if it to be used or not.
 
 for new connections, configData will be  {"customerId": "1234567890", "loginCustomerId": "12345"} and no key `subAccount` in Config so use login customerId

Comment (aashishmalik) [reply src/v0/destinations/google_adwords_enhanced_conversions/transform.js]: `{
                rudderAccountId: '25u5whFH7gVTnCiAjn4ykoCLGoC',
                listOfConversions: ['Page View', 'Product Added'],
                authStatus: 'active',
                subAccount: false,
                configData: '{"customerId": "1234567890"}',
                customerId: '1234567890',
                loginCustomerId: '12345',
              },`
              
updated code, once migration is done will remove old code that is kept as to be backward compatible

File: src/v0/destinations/google_adwords_enhanced_conversions/transform.js [hunk 2]
```diff
@@ -45,8 +45,9 @@ const responseBuilder = async (metadata, message, { Config }, payload) => {
   let { customerId, loginCustomerId } = Config;
   const { configData } = Config;
 
+  let configDetails = {};
   if (isDefinedAndNotNull(configData)) {
-    const configDetails = JSON.parse(configData);
+    configDetails = JSON.parse(configData);
     customerId = configDetails.customerId;
     if (isDefined(configDetails.loginCustomerId)) {
```
Comment (krishna2020) [inline src/v0/destinations/google_adwords_enhanced_conversions/transform.js:52]: this code piece is not needed

File: src/v0/destinations/google_adwords_enhanced_conversions/transform.js [hunk 3]
```diff
@@ -45,8 +45,9 @@ const responseBuilder = async (metadata, message, { Config }, payload) => {
   let { customerId, loginCustomerId } = Config;
   const { configData } = Config;
 
+  let configDetails = {};
```
Comment (krishna2020) [inline src/v0/destinations/google_adwords_enhanced_conversions/transform.js:48]: why `let` --> you are not updating the variable twice

File: src/v0/destinations/google_adwords_enhanced_conversions/transform.js [hunk 4]
```diff
@@ -70,6 +71,7 @@ const responseBuilder = async (metadata, message, { Config }, payload) => {
     'developer-token': getValueFromMessage(metadata, 'secret.developer_token'),
   };
   response.params = { event, customerId: filteredCustomerId };
+
   if (subAccount) {
```
Comment (krishna2020) [inline src/v0/destinations/google_adwords_enhanced_conversions/transform.js:75]: here add additional condition to execute this logic only if `!isDefined(configDetails.loginCustomerId)`

Comment (aashishmalik) [reply src/v0/destinations/google_adwords_enhanced_conversions/transform.js]: added

File: src/v0/destinations/google_adwords_enhanced_conversions/transform.js (hunk 3 above)
Comment (aashishmalik) [reply src/v0/destinations/google_adwords_enhanced_conversions/transform.js]: earlier was reassigning it so used let, now using ternary condition and const

**Comment by krishna2020** [comment]
This is no longer required as design got updated now without using cascade component

**Pull request closed without merging**